src/libraries/System.Text.Json/src/System/Text/Json/Serialization/Converters/Collection/DictionaryDefaultConverter.cs

```diff
@@ -197,7 +198,11 @@ protected static JsonConverter<TKey> GetKeyConverter(Type keyType, JsonSerialize
                             }
                         }
 
-                        state.Current.DictionaryKey = ReadDictionaryKey(ref reader, ref state);
```

Comment:
@steveharter, @layomia Does `state.Current.DictionaryKey` need to be set before `return true` at the end of the method?

https://github.com/dotnet/runtime/blob/9bb4bfe7b84ce0e05e55059fd3ab99448176d5ac/src/libraries/System.Text.Json/src/System/Text/Json/Serialization/Converters/Collection/DictionaryDefaultConverter.cs#L231-L233

For some sequence of branch executions, it was being set before, but now it isn't. I am not sure if it is necessary or mattered though. It is being set for all `return false`.

Reply:
No it only needs to be set for failure\false conditions, so it can remember the key on continuation (after the Stream reads the dictionary value).